daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/NodeInfo.scala

```diff
@@ -410,15 +410,15 @@ object NodeInfo extends Enum {
     Byte,
     Short,
     Long,
-    Integer,
-    Decimal,
     UnsignedInt,
     UnsignedByte,
     UnsignedShort,
     UnsignedLong,
     NonNegativeInteger,
+    Integer,
     Double,
     Float,
+    Decimal,
     HexBinary,
```

Comment:
So the order of this list is important? I think it would be helpful in the commit message what the problem was so looking back on this we have a better idea of why this change was made. Might also be useful to add a comment to explain why order is important.

I'd guess this is because PrimType.fromNodeInfo searches allPrimTypes looking for the first subTypeOf, and I guess some NodeInfo's are subtypes of multiple PrimTypes? So these need to be ordred from most specific to least specific, or somethign like that?

Reply:
Yes, I meant to add something along those lines to the commit message, but I pushed this up right before going to sleep and my groggy brain didn't remember.
allPrimTypes definitely needs to be in order or most specific to least specific, as we are doing a find(isSubTypeof(_)), which simply returns the first successful match.

Reply:
Also, looking at this with a clearer head this morning, there are a few other changes necessary. I'm going to push up a small revision with everything in appropriate order.